meja/parsetypes.ml

```diff
@@ -1,15 +1,31 @@
 type str = string Location.loc
 
-type pattern = PVariable of str
+type type_expr = {mutable desc: type_desc; id: int}
```

Comment:
It would be nice to avoid mutability if possible -- can you speak a bit about why you reached for it here?

Reply:
It's possible, I just leant on mutability here to make the typechecking code cleaner, simpler, and faster.

I'll rework it to avoid mutability.

Reply:
I don't have a huge problem with mutability in the type checker for what it's worth if it makes the code better.

Reply:
I agree with @imeckler.